QuizApp/Quiz/QuizViewModel.swift

```diff
@@ -27,17 +28,27 @@ class QuizViewModel {
                   description: "Movie Geography description that can usually span over multiple lines",
                   difficulty: "NORMAL", id: 7, imageUrl: "", name: "Mars", numberOfQuestions: 8)]
 
-    init(router: AppRouterProtocol) {
+    init(router: AppRouterProtocol, useCase: QuizUseCaseProtocol) {
         self.router = router
+        self.useCase = useCase
     }
 
     func getAllQuizzes() {
-        mockQuizzes.removeAll()
-        mockQuizzes = quizzes
+        quizzes = mockquizzes
     }
-    func getQuizzes(for category: String) {
-        mockQuizzes.removeAll()
-        mockQuizzes = quizzes.filter { $0.category == category }
+
+    @MainActor
+    func getQuizzes(for category: String, completion: @escaping () -> Void) {
```

Comment:
remove completion block and rely on @Published var (like you did for UserViewModel)